File: app/src/main/java/com/incendiary/androidboilerplate/data/local/DatabaseHelper.java
package com.incendiary.androidboilerplate.data.local;

import android.database.sqlite.SQLiteDatabase;
import com.incendiary.androidboilerplate.data.model.Ribot;
import com.squareup.sqlbrite.BriteDatabase;
import com.squareup.sqlbrite.SqlBrite;
import java.util.Collection;
import java.util.List;
import javax.inject.Inject;
import javax.inject.Singleton;
import rx.Observable;
import rx.Subscriber;

@Singleton public class DatabaseHelper {

  private final BriteDatabase mDb;

  @Inject public DatabaseHelper(DbOpenHelper dbOpenHelper) {
    mDb = SqlBrite.create().wrapDatabaseHelper(dbOpenHelper);
  }

  public BriteDatabase getBriteDb() {
    return mDb;
  }

  public Observable<Ribot> setRibots(final Collection<Ribot> newRibots) {
    return Observable.create(new Observable.OnSubscribe<Ribot>() {
      @Override public void call(Subscriber<? super Ribot> subscriber) {
        if (subscriber.isUnsubscribed()) return;
        BriteDatabase.Transaction transaction = mDb.newTransaction();

        try {
          mDb.delete(Db.RibotProfileTable.INSTANCE.getTABLE_NAME(), null);
          for (Ribot ribot : newRibots) {
            long result = mDb.insert(Db.RibotProfileTable.INSTANCE.getTABLE_NAME(),
                Db.RibotProfileTable.INSTANCE.toContentValues(ribot.profile()),
                SQLiteDatabase.CONFLICT_REPLACE);
            if (result >= 0) subscriber.onNext(ribot);
          }

          transaction.markSuccessful();
          subscriber.onCompleted();
        } finally {
          transaction.end();
        }
      }
    });
  }

  public Observable<List<Ribot>> getRibots() {
    return mDb.createQuery(Db.RibotProfileTable.INSTANCE.getTABLE_NAME(),
        "SELECT * FROM " + Db.RibotProfileTable.INSTANCE.getTABLE_NAME())
        .mapToList(cursor -> Ribot.create(Db.RibotProfileTable.INSTANCE.parseCursor(cursor)));
  }
}

File: app/src/main/java/com/incendiary/androidboilerplate/data/local/DatabaseHelper.kt
package com.incendiary.androidboilerplate.data.local

import android.database.sqlite.SQLiteDatabase
import com.incendiary.androidboilerplate.data.model.Ribot
import com.squareup.sqlbrite.BriteDatabase
import com.squareup.sqlbrite.SqlBrite
import rx.Observable
import javax.inject.Inject
import javax.inject.Singleton

@Singleton class DatabaseHelper
@Inject constructor(dbOpenHelper: DbOpenHelper) {

	val briteDb: BriteDatabase
			by lazy { SqlBrite.create().wrapDatabaseHelper(dbOpenHelper) }

	fun setRibots(newRibots: Collection<Ribot>): Observable<Ribot> {
		return Observable.create(Observable.OnSubscribe<Ribot> { subscriber ->
			if (subscriber.isUnsubscribed) return@OnSubscribe
			val transaction = briteDb.newTransaction()

			try {
				briteDb.delete(Db.RibotProfileTable.TABLE_NAME, null)
				for (ribot in newRibots) {
					val result = briteDb.insert(Db.RibotProfileTable.TABLE_NAME,
							Db.RibotProfileTable.toContentValues(ribot.profile),
							SQLiteDatabase.CONFLICT_REPLACE)
					if (result >= 0) subscriber.onNext(ribot)
				}

				transaction.markSuccessful()
				subscriber.onCompleted()
			} finally {
				transaction.end()
			}
		})
	}

	val ribots: Observable<List<Ribot>>
		get() = briteDb.createQuery(Db.RibotProfileTable.TABLE_NAME,
				"SELECT * FROM " + Db.RibotProfileTable.TABLE_NAME).mapToList { cursor -> Ribot(Db.RibotProfileTable.parseCursor(cursor)) }
}
